### ISSUE TITLE
VS 14.0 msvc headers aren't automatically detected

### ISSUE THREAD

**Issue by Clinkz24** [microsoft/vscode-cpptools#7789]
Bug type: Language Service

**Describe the bug**
- OS and Version: windows 10 and visual studio 2015
- VS Code Version: 1.57.1(user setup)
- C/C++ Extension Version:1.5.0-insiders3
- Other extensions you installed (and if the issue persists after disabling them):NO
- If using SSH remote, specify OS of remote machine:NO
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. C/C++ Extension update to 1.5.0-insiders3
2. restart my vscode
3. output unable to parse configuration using compilerPath
4. IntelliSense error like can not find vc include files

**Expected behavior**

C/C++ Extension 1.4.1 work well。

**Code sample and logs**

- Code sample

- Configurations in `c_cpp_properties.json`
```json
{
    "configurations": [
        {
            "name": "Win32",
            "defines": [
                "_DEBUG"
            ],
            "windowsSdkVersion": "10.0.10240.0",
            "cStandard": "c17",
            "cppStandard": "c++14",
            "intelliSenseMode": "msvc-x86",
            "compileCommands": ".vscode/compile_commands.json"
        }
    ],
    "version": 4
}
```
compile_commands.json:
```json
[
{
  "directory": "D:\\workspace\\dev\\gitlab\\IudgeServiceModules\\iudgeservicemodules",
  "arguments": ["D:\\workspace\\soft\\visualstudio\\Microsoft Visual Studio 14.0\\VC\\BIN\\cl.exe", "-c", "/EHsc", "-nologo", "-Zi", "-FS", "-Fdbuild\\ServiceMoudle4ChannelManager\\windows\\x86\\release\\compile.ServiceMoudle4ChannelManager.pdb", "-Ox", "-fp:fast", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\boost.pkg\\v1.68.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\tinyxml2.pkg\\v6.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\protobuf.pkg\\v3.6.1\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\cpp_redis.pkg\\v4.3.1\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\isc.pkg\\v1.4.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_utils.pkg\\v1.1.4\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\isf.pkg\\v2.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\icf.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\ds_access.pkg\\v1.3.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\dmodule_quotes.pkg\\v1.0.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\dmodule_channelmanager.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_netcore.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_loger.pkg\\v1.0.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\cryptopp.pkg\\v7.0.static\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\nlohmann_json.pkg\\v3.3.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\curl.pkg\\v6.2.0\\inc", "-DBOOST_ALL_DYN_LINK", "-MD", "-DNDEBUG", "-Fobuild\\.objs\\ServiceMoudle4ChannelManager\\windows\\x86\\release\\Moudules\\ServiceMoudle4ChannelManager\\src\\channel_manager.cpp.obj", "Moudules\\ServiceMoudle4ChannelManager\\src\\channel_manager.cpp"],
  "file": "Moudules\\ServiceMoudle4ChannelManager\\src\\channel_manager.cpp"
},
{
  "directory": "D:\\workspace\\dev\\gitlab\\IudgeServiceModules\\iudgeservicemodules",
  "arguments": ["D:\\workspace\\soft\\visualstudio\\Microsoft Visual Studio 14.0\\VC\\BIN\\cl.exe", "-c", "/EHsc", "-nologo", "-Zi", "-FS", "-Fdbuild\\ServiceMoudle4ChannelManager\\windows\\x86\\release\\compile.ServiceMoudle4ChannelManager.pdb", "-Ox", "-fp:fast", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\boost.pkg\\v1.68.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\tinyxml2.pkg\\v6.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\protobuf.pkg\\v3.6.1\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\cpp_redis.pkg\\v4.3.1\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\isc.pkg\\v1.4.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_utils.pkg\\v1.1.4\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\isf.pkg\\v2.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\icf.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\ds_access.pkg\\v1.3.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\dmodule_quotes.pkg\\v1.0.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\dmodule_channelmanager.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_netcore.pkg\\v1.2.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\iudge\\iudge_loger.pkg\\v1.0.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\cryptopp.pkg\\v7.0.static\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\nlohmann_json.pkg\\v3.3.0\\inc", "-ID:\\workspace\\dev\\gitlab\\IudgeBasicService\\iudgebasicservice\\pkgs-local\\third\\curl.pkg\\v6.2.0\\inc", "-DBOOST_ALL_DYN_LINK", "-MD", "-DNDEBUG", "-Fobuild\\.objs\\ServiceMoudle4ChannelManager\\windows\\x86\\release\\Moudules\\ServiceMoudle4ChannelManager\\src\\ctrlconn_handler.cpp.obj", "Moudules\\ServiceMoudle4ChannelManager\\src\\ctrlconn_handler.cpp"],
  "file": "Moudules\\ServiceMoudle4ChannelManager\\src\\ctrlconn_handler.cpp"
}
]
```
- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette
```log
-------- Diagnostics - 2021/7/7下午9:01:37
Version: 1.5.0-insiders3
Current Configuration:
{
    "name": "Win32",
    "defines": [
        "_DEBUG"
    ],
    "windowsSdkVersion": "10.0.10240.0",
    "cStandard": "c17",
    "cppStandard": "c++14",
    "intelliSenseMode": "msvc-x86",
    "compileCommands": ".vscode/compile_commands.json",
    "compilerPathIsExplicit": false,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerArgs": [],
    "browse": {
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ D:\workspace\dev\gitlab\IudgeServiceModules\iudgeservicemodules\Moudules\ServiceMoudle4ChannelManager\src\channel_manager.cpp ]:
    D:\WORKSPACE\DEV\GITLAB\IUDGESERVICEMODULES\IUDGESERVICEMODULES\MOUDULES\SERVICEMOUDLE4CHANNELMANAGER\SRC\CHANNEL_MANAGER.CPP
Translation Unit Configurations:
[ D:\workspace\dev\gitlab\IudgeServiceModules\iudgeservicemodules\Moudules\ServiceMoudle4ChannelManager\src\channel_manager.cpp ]:
    Process ID: 13044
    Memory Usage: 233 MB
    Includes:
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\BOOST.PKG\V1.68.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\TINYXML2.PKG\V6.2.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\PROTOBUF.PKG\V3.6.1\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CPP_REDIS.PKG\V4.3.1\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ISC.PKG\V1.4.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_UTILS.PKG\V1.1.4\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ISF.PKG\V2.2.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ICF.PKG\V1.2.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DS_ACCESS.PKG\V1.3.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DMODULE_QUOTES.PKG\V1.0.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DMODULE_CHANNELMANAGER.PKG\V1.2.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_NETCORE.PKG\V1.2.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_LOGER.PKG\V1.0.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CRYPTOPP.PKG\V7.0.STATIC\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\NLOHMANN_JSON.PKG\V3.3.0\INC
        D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CURL.PKG\V6.2.0\INC
    Defines:
        BOOST_ALL_DYN_LINK
        NDEBUG
    Standard Version: ms_c++14
    IntelliSense Mode: windows-msvc-x86
    compile_commands.json entry:
        directory: D:\workspace\dev\gitlab\IudgeServiceModules\iudgeservicemodules
        file: Moudules\ServiceMoudle4ChannelManager\src\channel_manager.cpp
        arguments:
            D:\workspace\soft\visualstudio\Microsoft Visual Studio 14.0\VC\BIN\cl.exe
            -c
            /EHsc
            -nologo
            -Zi
            -FS
            -Fdbuild\ServiceMoudle4ChannelManager\windows\x86\release\compile.ServiceMoudle4ChannelManager.pdb
            -Ox
            -fp:fast
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\boost.pkg\v1.68.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\tinyxml2.pkg\v6.2.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\protobuf.pkg\v3.6.1\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\cpp_redis.pkg\v4.3.1\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\isc.pkg\v1.4.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\iudge_utils.pkg\v1.1.4\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\isf.pkg\v2.2.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\icf.pkg\v1.2.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\ds_access.pkg\v1.3.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\dmodule_quotes.pkg\v1.0.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\dmodule_channelmanager.pkg\v1.2.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\iudge_netcore.pkg\v1.2.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\iudge\iudge_loger.pkg\v1.0.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\cryptopp.pkg\v7.0.static\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\nlohmann_json.pkg\v3.3.0\inc
            -ID:\workspace\dev\gitlab\IudgeBasicService\iudgebasicservice\pkgs-local\third\curl.pkg\v6.2.0\inc
            -DBOOST_ALL_DYN_LINK
            -MD
            -DNDEBUG
            -Fobuild\.objs\ServiceMoudle4ChannelManager\windows\x86\release\Moudules\ServiceMoudle4ChannelManager\src\channel_manager.cpp.obj
            Moudules\ServiceMoudle4ChannelManager\src\channel_manager.cpp
Total Memory Usage: 233 MB
Browse Paths from compile_commands.json, from workspace folder: D:\workspace\dev\gitlab\IudgeServiceModules\iudgeservicemodules
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DMODULE_CHANNELMANAGER.PKG\V1.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DMODULE_QUOTES.PKG\V1.0.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\DS_ACCESS.PKG\V1.3.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ICF.PKG\V1.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ISC.PKG\V1.4.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\ISF.PKG\V2.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_LOGER.PKG\V1.0.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_NETCORE.PKG\V1.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\IUDGE\IUDGE_UTILS.PKG\V1.1.4\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\BOOST.PKG\V1.68.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CPP_REDIS.PKG\V4.3.1\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CRYPTOPP.PKG\V7.0.STATIC\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\CURL.PKG\V6.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\NLOHMANN_JSON.PKG\V3.3.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\PROTOBUF.PKG\V3.6.1\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGEBASICSERVICE\IUDGEBASICSERVICE\PKGS-LOCAL\THIRD\TINYXML2.PKG\V6.2.0\INC
    D:\WORKSPACE\DEV\GITLAB\IUDGESERVICEMODULES\IUDGESERVICEMODULES\MOUDULES\SERVICEMOUDLE4CHANNELMANAGER\CONFIGMANAGER
    D:\WORKSPACE\DEV\GITLAB\IUDGESERVICEMODULES\IUDGESERVICEMODULES\MOUDULES\SERVICEMOUDLE4CHANNELMANAGER\SRC
    D:\WORKSPACE\DEV\GITLAB\IUDGESERVICEMODULES\IUDGESERVICEMODULES\MOUDULES\SERVICEMOUDLE4HTTP2ITP\SRC

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 14272


```
- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Screenshots**

![image](https://user-images.githubusercontent.com/15918670/124763756-e36a8800-df66-11eb-8be4-95a947ddc8d7.png)

![image](https://user-images.githubusercontent.com/15918670/124763853-fc733900-df66-11eb-8b7f-444ed3f55983.png)

![image](https://user-images.githubusercontent.com/15918670/124764015-1d3b8e80-df67-11eb-81c3-0f08f4a49438.png)

**Additional context**

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. We believe we know the cause. 1.5.0-insiders2 should also not repro the issue https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.0-insiders2 .

**Comment by sean-mcmanus** [comment]
Fixed with 1.5.0 (https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.0).

**Comment by Clinkz24** [comment]
thanks a lot😁😁
